extensions/gmsa-dc/v1/Setup-gMSA.ps1

```diff
@@ -0,0 +1,271 @@
+    if ($AdditionalAccounts) {
+        $AdditionalAccounts | ForEach-Object {
+            $output.ActiveDirectoryConfig.GroupManagedServiceAccounts += @{"Name" = $_.AccountName; "Scope" = $_.Domain }
+        }
+    }
+    
+    # Create CmsPlugins Object
+    $output.CmsPlugins = @("ActiveDirectory")
+
+    # Create DomainJoinConfig Object
+    $output.DomainJoinConfig = @{}
+    $output.DomainJoinConfig.DnsName = $ADDomain.DNSRoot
+    $output.DomainJoinConfig.Guid = $ADDomain.ObjectGUID
+    $output.DomainJoinConfig.DnsTreeName = $ADDomain.Forest
+    $output.DomainJoinConfig.NetBiosName = $ADDomain.NetBIOSName
+    $output.DomainJoinConfig.Sid = $ADDomain.DomainSID.Value
+    $output.DomainJoinConfig.MachineAccountName = $AccountName
+
+    $output | ConvertTo-Json -Depth 5 | Out-File -FilePath $FullPath -Encoding ascii -NoClobber:$NoClobber
+	
+	Install-Module powershell-yaml -Force
+
+    $CredSpecJson = Get-Item $FullPath | Select-Object @{
+        Name       = 'Name'
+        Expression = { $_.Name }
+    },
+    @{
+        Name       = 'Path'
+        Expression = { $_.FullName }
+    }
+	
+	# This section of code borrowed from https://github.com/kubernetes-sigs/windows-gmsa/blob/master/scripts/GenerateCredentialSpecResource.ps1
+    $dockerCredSpecPath = $CredSpecJson.Path
+    Sleep 2
+    $credSpecContents = Get-Content $dockerCredSpecPath | ConvertFrom-Json
+    $ManifestFile = "gmsa-cred-spec-gmsa-e2e.yml"
+	
+	# generate the k8s resource
+    $resource = [ordered]@{
+        "apiVersion" = "windows.k8s.io/v1alpha1";
+        "kind" = 'GMSACredentialSpec';
+        "metadata" = @{
+        "name" = $AccountName
+    };
+    "credspec" = $credSpecContents
+}
+
+ConvertTo-Yaml $resource | Set-Content $ManifestFile
+
+Write-Output "K8S manifest rendered at $ManifestFile"
+}
+
+# Change to working directory
+Set-Location -Path $GMSARoot
+
+# Script Logging - Feel free to comment out if you like.
+Start-Transcript -Path "$GMSARoot\Setup-gmsa.txt" -Append
+# Making sure the AD services are up and running
+do {$ADresult = CheckADServices} while ($ADresult -eq $false)
+
+# Import Active Directory PowerShell module
+do {$ADPSresult = ImportADPS} while ($ADPSresult -eq $false)
+
+# Make sure AD server instance is listening on port
+do {$CGresult = CheckGroup} while ($CGresult -eq $false)
+
+
+if (Get-AdGroupMember -Identity "Enterprise Admins" | Select-String -Pattern "gmsa-admin" -Quiet) {
+    
+	# Add the KDS Root Key
+    $KdsRootKey = Get-KdsRootKey
+    if ($KdsRootKey -eq $null) {
+		# This command creates the KDS Root Key with a time 10 hours in the past.
+		# Active Directory will not permit use of this key until at least 10 hours has
+		# passed to allow distribution to all Domain Controllers in the domain.
+		# https://docs.microsoft.com/en-us/windows-server/security/group-managed-service-accounts/create-the-key-distribution-services-kds-root-key
```

Comment:
nit: the indentation here looks off for the comment blocks. Please check the indentation.